internal/controllers/nmc_reconciler_test.go

```diff
@@ -1158,39 +1184,143 @@ var _ = Describe("nmcReconcilerHelperImpl_RemovePodFinalizers", func() {
+		client.EXPECT().Get(ctx, gomock.Any(), gomock.Any(), gomock.Any()).Return(fmt.Errorf("some error"))
+		err := client.Get(ctx, types.NamespacedName{Name: nmc.Name}, &v1.Node{})
+		Expect(err).To(HaveOccurred())
+	})
+	It("Should fail patching node after change in labels", func() {
+		node := v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					fmt.Sprintf("kmm.node.kubernetes.io/%s.%s.ready", nsFirst, nameFirst): "",
+				},
+				Name: nodeName,
+			},
+		}
+		emptySet := sets.Set[types.NamespacedName]{}
+		emptyMap := map[types.NamespacedName]kmmv1beta1.ModuleConfig{}
+
+		gomock.InOrder(
+			mlph.EXPECT().getNodeKernelModuleReadyLabels(node).Return(emptySet),
+			mlph.EXPECT().getDeprecatedKernelModuleReadyLabels(node).Return(sets.Set[string]{}),
+			mlph.EXPECT().getSpecLabelsAndTheirConfigs(&nmc).Return(emptyMap),
+			mlph.EXPECT().getStatusLabelsAndTheirConfigs(&nmc).Return(emptyMap),
+			mlph.EXPECT().removeOrphanedLabels(emptySet, emptyMap, emptyMap).Return([]types.NamespacedName{{Name: nameSecond, Namespace: nsSecond}}),
+			mlph.EXPECT().addEqualLabels(emptySet, emptyMap, emptyMap).Return([]types.NamespacedName{{Name: nameFirst, Namespace: nsFirst}}),
+			n.EXPECT().
+				UpdateLabels(
+					ctx,
+					&node,
+					[]string{fmt.Sprintf("kmm.node.kubernetes.io/%s.%s.ready", nsFirst, nameFirst)},
+					[]string{fmt.Sprintf("kmm.node.kubernetes.io/%s.%s.ready", nsSecond, nameSecond)},
+				).Return(fmt.Errorf("some error")),
+		)
+		_, _, err := wh.UpdateNodeLabels(ctx, &nmc, &node)
+		Expect(err).To(HaveOccurred())
+	})
+	It("Should work as expected", func() {
+		node := v1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					fmt.Sprintf("kmm.node.kubernetes.io/%s.%s.ready", nsFirst, nameFirst): "",
+					fmt.Sprintf("kmm.node.kubernetes.io/%s.%s.ready", nsFirst, nameFirst): "",
+				},
+				Name: nodeName,
+			},
+		}
+		emptySet := sets.Set[types.NamespacedName]{}
+		emptyMap := map[types.NamespacedName]kmmv1beta1.ModuleConfig{}
+
+		gomock.InOrder(
+			mlph.EXPECT().getNodeKernelModuleReadyLabels(node).Return(emptySet),
+			mlph.EXPECT().getDeprecatedKernelModuleReadyLabels(node).Return(sets.Set[string]{}),
+			mlph.EXPECT().getSpecLabelsAndTheirConfigs(&nmc).Return(emptyMap),
+			mlph.EXPECT().getStatusLabelsAndTheirConfigs(&nmc).Return(emptyMap),
+			mlph.EXPECT().removeOrphanedLabels(emptySet, emptyMap, emptyMap).Return([]types.NamespacedName{{Name: nameSecond, Namespace: nsSecond}}),
+			mlph.EXPECT().addEqualLabels(emptySet, emptyMap, emptyMap).Return([]types.NamespacedName{{Name: nameFirst, Namespace: nsFirst}}),
+			n.EXPECT().
+				UpdateLabels(
+					ctx,
+					&node,
+					[]string{fmt.Sprintf("kmm.node.kubernetes.io/%s.%s.ready", nsFirst, nameFirst)},
+					[]string{fmt.Sprintf("kmm.node.kubernetes.io/%s.%s.ready", nsSecond, nameSecond)},
+				).Return(nil),
+		)
+		_, _, err := wh.UpdateNodeLabels(ctx, &nmc, &node)
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
This is only testing that no error was returned. What about the part that validates that the node was updated correctly?

Reply:
I added another Except for the new node label (next line)